### PR TITLE
19332 - Short name account link modal

### PR DESCRIPTION
**Pull request by rodrigo-barraza**
*Issue #:*
https://github.com/bcgov/entity/issues/<Put the github issue number here>

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/pay/UnlinkedShortNameTable.vue [hunk 1]
```diff
@@ -251,6 +246,10 @@ export default defineComponent({
       }
     ]
 
+    function selectAccount (account: any) {
```
Comment (seeker25) [inline auth-web/src/components/pay/UnlinkedShortNameTable.vue:249]: `any` probably should fix the typing plus return types if possible

File: auth-web/src/components/pay/UnlinkedShortNameTable.vue [hunk 2]
```diff
@@ -279,6 +278,19 @@ export default defineComponent({
       accountLinkingDialog.value.close()
     }
 
+    async function linkAccount () {
+      if (state.selectedShortName?.id && state.selectedAccount?.id) {
```
Comment (seeker25) [inline auth-web/src/components/pay/UnlinkedShortNameTable.vue:282]: probably easier to reverse this and just return

```
if (!state.selectedShortName?.id || !state.selectedAccount?.id) {
   return
}
```

File: auth-web/src/components/pay/ShortNameLookup.vue [hunk 3]
```diff
@@ -156,95 +107,115 @@ export default defineComponent({
...
+      color: black;
+    }
+  }
+}
+
+.undo {
+  padding: 0;
+  margin: 0;
+  height: 22px;
+  display: flex;
```
Comment (seeker25) [inline auth-web/src/components/pay/ShortNameLookup.vue:212]: vuetify styling classes for a few of these

File: auth-web/src/components/pay/ShortNameLookup.vue [hunk 4]
```diff
@@ -1,153 +1,104 @@
...
-      >
-        <v-row class="business-lookup-result pt-1">
+      <template #item="{ item }">
+        <v-row class="short-name-lookup-result pt-1">
           <v-col
             cols="3"
             class="result-identifier d-inline-flex"
           >
-            {{ item.nrNum }}
+            {{ item.accountId }}2
```
Comment (seeker25) [inline auth-web/src/components/pay/ShortNameLookup.vue:63]: Extra 2 here

File: auth-web/src/components/auth/common/ModalDialog.vue [hunk 5]
```diff
@@ -194,4 +194,10 @@ export default defineComponent({
   [title] {
     text-align: center;
   }
+  .short-name-lookup-dialog {
+    background: red;
```
Comment (seeker25) [inline auth-web/src/components/auth/common/ModalDialog.vue:198]: Probably a way to do this inline? Also why is there short name specific code in here? it should be generic

**Review by seeker25 — COMMENTED** [review]
see above

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2720-mgg2g9o6.web.app

**Comment by seeker25** [comment]
Modal size needs to be increased, should be 720x, I think it's 620 ish

**Comment by seeker25** [comment]
image

**Comment by seeker25** [comment]
button alignment and styling needs a tweak:

image

**Comment by seeker25** [comment]
Close -> Cancel it should be outlined in style

**Comment by seeker25** [comment]
spacing here:
image

**Comment by seeker25** [comment]
image

**Comment by seeker25** [comment]
I tried to link, 
image

**Comment by seeker25** [comment]
I tried again, pressed it, it switched tabs but didnt' close the dialog:

image

**Comment by seeker25** [comment]
Also think you need to handle `"EFT_SHORT_NAME_ALREADY_MAPPED"` error message

File: auth-web/src/components/pay/ShortNameLookup.vue [hunk 6]
```diff
@@ -156,95 +107,116 @@ export default defineComponent({
...
+      }
     }, 600)
 
     watch(() => states.searchField, onSearchFieldChanged)
 
-    const onItemSelected = (input: BusinessLookupResultIF | NameRequestLookupResultIF) => {
-      if (input) {
-        emit(props.lookupType, input)
+    const onItemSelected = (account: EFTShortnameResponse) => {
+      console.log(123, account)
```
Comment (seeker25) [inline auth-web/src/components/pay/ShortNameLookup.vue:169]: `console.log(123`

File: auth-web/src/assets/scss/ModalDialog.scss [hunk 7]
```diff
@@ -1,7 +1,7 @@
 .v-dialog {
   > .v-card {
     > div {
-        padding: 16px 40px 0 40px;
+        padding: 32px 40px 0 40px;
```
Comment (seeker25) [inline auth-web/src/assets/scss/ModalDialog.scss:4]: This changes all modal dialogs, wouldn't suggest doing this

File: auth-web/src/components/pay/UnlinkedShortNameTable.vue [hunk 8]
```diff
@@ -251,6 +247,10 @@ export default defineComponent({
       }
     ]
 
+    function selectAccount (account: EFTShortnameResponse) {
+      state.selectedAccount = account
```
Comment (seeker25) [inline auth-web/src/components/pay/UnlinkedShortNameTable.vue:251]: might be able to change
@account="selectAccount"

to
 `@account=state.selectedAccount = $event`

File: auth-web/src/models/pay/short-name.ts [hunk 9]
```diff
@@ -63,6 +64,7 @@ export interface UnlinkedShortNameState {
   dateRangeReset: number
   clearFiltersTrigger: number
   selectedShortName: object
+  selectedAccount: object
```
Comment (seeker25) [inline auth-web/src/models/pay/short-name.ts:67]: probably better defining the object, I saw some warnings complaining about the object type

File: auth-web/src/components/auth/common/ModalDialog.vue (hunk 5 above)
Comment (seeker25) [reply auth-web/src/components/auth/common/ModalDialog.vue:198]: Alternatively, rename the selector to `lookup-dialog`, I think that would work too

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2720-mgg2g9o6.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2720-mgg2g9o6.web.app

**Review by seeker25 — COMMENTED** [review]
Could you update package + package-lock.json again please?

**Review by seeker25 — APPROVED** [review]

**Pull request merged**